File: app/src/main/java/com/task/data/DataRepository.java
package com.task.data;

import com.task.data.local.LocalRepository;
import com.task.data.remote.RemoteRepository;
import com.task.data.remote.dto.NewsModel;

import javax.inject.Inject;

import io.reactivex.Single;


/**
 * Created by AhmedEltaher on 5/12/2016
 */

public class DataRepository implements DataSource {
    private RemoteRepository remoteRepository;
    private LocalRepository localRepository;

    @Inject
    public DataRepository(RemoteRepository remoteRepository, LocalRepository localRepository) {
        this.remoteRepository = remoteRepository;
        this.localRepository = localRepository;
    }

    @Override
    public Single<NewsModel> requestNews() {
        return remoteRepository.getNews();
    }
}

File: app/src/main/java/com/task/data/DataRepository.kt
package com.task.data

import com.task.data.local.LocalData
import com.task.data.remote.RemoteData
import com.task.data.remote.dto.NewsModel
import com.task.utils.wrapEspressoIdlingResource
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.flow
import kotlinx.coroutines.flow.flowOn
import javax.inject.Inject


/**
 * Created by AhmedEltaher
 */

class DataRepository @Inject
constructor(private val remoteRepository: RemoteData, private val localRepository: LocalData) : DataRepositorySource {

    override suspend fun requestNews(): Flow<Resource<NewsModel>> {
        return flow {
//                emit(Resource.Loading())
                emit(remoteRepository.requestNews())
            }.flowOn(Dispatchers.IO)
    }
}
